pipeline/splicing/calcVarPriors.py

```diff
@@ -139,22 +168,314 @@ def getVarType(variant):
+        del varExons["exon1"]
+    varStrand = getVarStrand(variant)
+    acceptorBoundaries = {}
+    for exon in varExons.keys():
+        exonStart = int(varExons[exon]["exonStart"])
+        if varStrand == "+":
+            acceptorStart = exonStart - 20 + 1
+            acceptorEnd = exonStart + 3
+        else:            
+            acceptorStart = exonStart + 20
+            acceptorEnd = exonStart - 3 + 1
+        acceptorBoundaries[exon] = {"acceptorStart": acceptorStart,
+                                    "acceptorEnd": acceptorEnd}
+
+    return acceptorBoundaries
+
+
+def varInExon(variant):
+    '''
+    Given a variant, determines if variant genomic position is inside transcript boundaries
+    AND if variant is in an exon
+    Returns true if variant is in an exon
+    '''
+    varGenPos = int(variant["Pos"])
+    varExons = getExonBoundaries(variant)
+    varStrand = getVarStrand(variant)
+    varOutBounds = varOutsideBoundaries(variant)
+    if varOutBounds == False:
+        for exon in varExons.keys():
+            exonStart = varExons[exon]["exonStart"]
+            exonEnd = varExons[exon]["exonEnd"]
+            if varStrand == "+":
+                if varGenPos > exonStart and varGenPos <= exonEnd:
+                    return True
+            else:
+                if varGenPos <= exonStart and varGenPos >= exonEnd:
+                    return True
+        return False
+    else:
+        return "varOutBounds"    
+
+    
+def varInSpliceDonor(variant):
+    '''
+    Given a variant, determines if a variant is in reference transcript's splice donor region
+    splice donor region = last 3 bases in exon and first 6 bases in intron
+    Returns True if variant in splice donor region
+    '''
+    varGenPos = int(variant["Pos"])
+    varStrand = getVarStrand(variant)
+    donorBounds = getRefSpliceDonorBoundaries(variant)
+    for exon in donorBounds.keys():
+        donorStart = donorBounds[exon]["donorStart"]
+        donorEnd = donorBounds[exon]["donorEnd"]
+        if varStrand == "+":
+            if varGenPos >= donorStart and varGenPos <= donorEnd:
+                return True
+        else:
+            if varGenPos <= donorStart and varGenPos >= donorEnd:
+                return True
+    return False
+
+
+def varInSpliceAcceptor(variant):
+    '''
+    Given a variant, determines if a variant is in reference transcript's splice acceptor region
+    splice acceptor region = 20 bases preceding exon and first 3 bases in exon
+    Returns True if variant in splice acceptor region
+    '''
+    varGenPos = int(variant["Pos"])
+    varStrand = getVarStrand(variant)
+    acceptorBounds = getRefSpliceAcceptorBoundaries(variant)
+    for exon in acceptorBounds.keys():
+        acceptorStart = acceptorBounds[exon]["acceptorStart"]
+        acceptorEnd = acceptorBounds[exon]["acceptorEnd"]
+        if varStrand == "+":
+            if varGenPos >= acceptorStart and varGenPos <= acceptorEnd:
+                return True
```

Comment:
return in for loop, double check

Reply:
Same as above comment but with splice acceptor regions instead of splice donor regions